src/gpu/locks.rs

```diff
@@ -0,0 +1,89 @@
+use fs2::FileExt;
+use log::{info, warn};
+use std::fs::File;
+use std::path::PathBuf;
+
+const GPU_LOCK_NAME: &str = "bellman.gpu.lock";
+const PRIORITY_LOCK_NAME: &str = "bellman.priority.lock";
+fn tmp_path(filename: &str) -> PathBuf {
+    let mut p = std::env::temp_dir();
+    p.push(filename);
+    p
+}
+
+#[derive(Debug)]
+pub struct GPULock(File);
+impl GPULock {
+    pub fn lock() -> GPULock {
+        info!("Acquiring GPU lock...");
+        let f = File::create(tmp_path(GPU_LOCK_NAME)).unwrap();
+        f.lock_exclusive().unwrap();
+        info!("GPU lock acquired!");
+        GPULock(f)
+    }
+}
+impl Drop for GPULock {
+    fn drop(&mut self) {
+        info!("GPU lock released!");
+    }
+}
+
+#[derive(Debug)]
+pub struct PriorityLock(File);
+impl PriorityLock {
+    pub fn lock_if_priority(priority: bool) -> Option<PriorityLock> {
```

Comment:
Rather than passing a bool, I would rather see an api with two different methods `lock_with_priority` and `lock_without_priority`

Reply:
hmm, actually it doesn't lock it when priority is false. it is just calling lock() when priority flag is true. we could also put that if condition inside create_proof like:
```
let _prio_lock = if priority { Some(PriorityLock::lock()) } else { None };
```
i thought it is cleaner to just put that flag check inside PriorityLock::lock() function